Improve description of cancellation process in workflow documentation (#44242)
github/docs: content/actions/reference/workflows-and-actions/workflow-cancellation.md

1. For jobs that need to be canceled, the server sends a cancellation message to all the runner machines with jobs that need to be canceled.
1. For jobs that continue to run, the server re-evaluates `if` conditions for the unfinished steps. If the condition evaluates to `true`, the step continues to run. You can use the `cancelled` expression to apply a status check of `cancelled()`. For more information, see [AUTOTITLE](/actions/reference/evaluate-expressions-in-workflows-and-actions#cancelled).
1. For steps that need to be canceled, the runner machine sends `SIGINT/Ctrl-C` to the step's entry process (`node` for JavaScript actions, `docker` for container actions, and `bash/cmd/pwd` when using `run` in a step). If the process doesn't exit within 7500 ms, the runner will send `SIGTERM/Ctrl-Break` to the process, then wait for 2500 ms for the process to exit. If the process is still running, the runner kills the process tree.

> 1. After the 5 minute cancellation timeout period, the server will forcibly terminate all jobs (marked in step 1 to be canceled) and steps that are still running.

Reviewer: ```suggestion
1. After the 5 minute cancellation timeout period, the server will forcibly terminate all jobs and steps marked for cancellation that are still running.
```

Just changing the wording so it's less specific, Copilot won't object, and if we change the steps at some point this wording won't have to be updated.
